### DISCUSSION TITLE
how to add new column in each maf dataframe?

### DISCUSSION THREAD

**Discussion by complexgenome** [PoisonAlien/maftools#1023] [Q&A] [answered]
Hi there,

I'm interested to visualize VAF in my cohort.

I don't have VAF column in my MAF files. I read that I can calculate using ratio of counts of ref and alt allele.
https://www.biostars.org/p/161298/#161777

I've following code:

```
maf_files = list.files(path = "edited_mafs_no_duplicates/", pattern = "*.\\.maf", full.names = TRUE)

mafs_datatable = lapply(X = maf_files, function(x){
  data.table::fread(file = x, sep = "\t", select = required.fields, skip = "Hugo_Symbol")
})

mafs = data.table::rbindlist(l = mafs_datatable, use.names = TRUE, fill = TRUE, idcol = "Source_MAF")

mafs_read = maftools::read.maf(maf = mafs)
```

How do I add or calculate this column to the list of data frames I've?

**Comment by PoisonAlien** [comment]
Hi,

It depends on what columns your maf has. If your ref and alt counts are stored under `t_ref_count` and `t_alt_count`, you could add a new `t_vaf` as in below

```r
mafs = data.table::rbindlist(l = mafs_datatable, use.names = TRUE, fill = TRUE, idcol = "Source_MAF")
mafs[,t_vaf := as.numeric(as.character(t_alt_count))/(as.numeric(as.character(t_ref_count)) + as.numeric(as.character(t_alt_count)))]
```

**Comment by complexgenome** [comment]
Hi @PoisonAlien 
Thank you for looking into this.
I'd like to confirm for the VAF:

1)
Can VAF not be, VAF=t_alt/((t_alt+t_ref)*2) since there are two copies.

2)
Or VAF be VAF=t_alt/(t_ref) that is count of alt divided by ref only.

3)
Your suggested way
VAF=t_alt/(t_alt+ t_ref)

Sorry this may see confusing.

**Reply by PoisonAlien** [reply, marked as answer]
Hi, 
It's the option 3. You would like to estimate the frequency of the variant allele of the total depth.

**Reply by complexgenome** [reply]
Thanks.

**Comment by NogitoErgoSum** [comment]
Hi,

I thought it was related: 

When we add a new column to a maf object, let's say I want to merge to columns: 

`maf@data@new_column <- paste(maf@data$Hugo_Symbol, maf@data$HGVSp_Short,sep = "_")`

If I then use subsetMaf with the newly made column like so:

`filtered_maf <- subsetMaf(maf, query = "new_column == 'CSF3R_p.D510H'")`

I get a "Object 'new_column' not found amongst [Hugo_Symbol, ..." error. When I check the maf object I can't see this new column in maf@maf.silent. What is the correct way of adding new columns to our maf?

Thank you
